api/wasm/cpp/proxy_wasm_externs.h

```diff
@@ -94,6 +94,8 @@ extern "C" WasmResult proxy_get_buffer_bytes(BufferType type, uint32_t start, ui
                                              const char** ptr, size_t* size);
 extern "C" WasmResult proxy_get_buffer_status(BufferType type, size_t* length_ptr,
                                               uint32_t* flags_ptr);
+extern "C" WasmResult proxy_set_buffer(BufferType type, uint32_t start, uint32_t length,
+                                       const char* ptr, size_t size, uint32_t flags);
```

Comment:
I'm not sure if this should be old-style `proxy_set_buffer_bytes` or new-style `proxy_set_buffer`? What do you think, @jplevyak?

Reply:
Let's be consistent and make this proxy_set_buffer_bytes and then do the ABI changes in one go.

Reply:
Done.

Should I also add a no-op `proxy_set_buffer_status` in order to include it in v0.1.0 ABI? (there is no easy way to emit `end_of_stream` in HTTP body, as far as I can tell).